core/templates/dev/head/pages/exploration_editor/ExplorationStatesService.js

```diff
@@ -99,6 +99,44 @@ oppia.factory('ExplorationStatesService', [
       widget_customization_args: ['interaction', 'customizationArgs']
     };
 
+    CONTENT_ID_EXTRACTION = {
+      answer_groups: function(answerGroups) {
+        contentIds = new Set();
+        answerGroups.forEach(function(answerGroup) {
+          contentIds.add(answerGroup.outcome.feedback.getContentId());
+        });
+        return contentIds;
+      },
+      default_outcome: function(defaultOutcome) {
+        contentIds = new Set();
+        if (defaultOutcome) {
+          contentIds.add(defaultOutcome.feedback.getContentId());
+        }
+        return contentIds;
+      },
+      hints: function(hints) {
+        contentIds = new Set();
+        hints.forEach(function(hint) {
+          contentIds.add(hint.hintContent.getContentId());
+        });
+        return contentIds;
+      },
+      solution: function(solution) {
+        contentIds = new Set();
+        if (solution) {
+          contentIds.add(solution.explanation.getContentId());
+        }
+        return contentIds;
+      }
+    };
+
+    var _getSetDifference = function(setA, setB) {
```

Comment:
Will _getElementsInFirstSetButNotInSecond fit? Avoids confusion with symmetric difference etc.

(If not, OK to leave as is.)

Reply:
Sounds good. Done!